syntax = "proto3";

package proto;

option go_package = "/proto";

service ImageProcessingService {rpc ProcessImage(stream ImageRequest) returns (stream ImageResponse);}

message ImageRequest {
  uint32 index = 1;
  bytes image = 2;
}

message ImageResponse {
  uint32 index = 1;
  bytes image_with_contour = 2;
  string recognition_result = 3;
  float metric = 4;
}